store/tikv/mock-tikv/executor.go

```diff
@@ -313,37 +334,37 @@ func handleRowData(columns []*tipb.ColumnInfo, colTps map[int64]*types.FieldType
 			if err1 != nil {
 				return nil, errors.Trace(err1)
 			}
-			values[col.GetColumnId()] = handleData
+			values[colIDs[id]] = handleData
 			continue
 		}
-		_, ok := values[col.GetColumnId()]
-		if ok {
+		if hasColVal(values, colIDs, id) {
 			continue
 		}
 		if len(col.DefaultVal) > 0 {
-			values[col.GetColumnId()] = col.DefaultVal
+			values[colIDs[id]] = col.DefaultVal
 			continue
 		}
 		if mysql.HasNotNullFlag(uint(col.GetFlag())) {
 			return nil, errors.New("Miss column")
 		}
-		values[col.GetColumnId()] = []byte{codec.NilFlag}
+		values[colIDs[id]] = []byte{codec.NilFlag}
 	}
 
 	return values, nil
 }
 
-// setColumnValueToEval puts column values into evaluator, the values will be used for expr evaluation.
-func setColumnValueToEval(eval *xeval.Evaluator, handle int64, row map[int64][]byte, cols map[int64]*tipb.ColumnInfo) error {
-	for colID, col := range cols {
+// setSchemaValueToEval puts column values into evaluator, the values will be used for expr evaluation.
+func setSchemaValueToEval(eval *xeval.Evaluator, handle int64, row [][]byte, cols []*tipb.ColumnInfo, colIDs map[int64]int) error {
```

Comment:
The columns count of the last argument is less than or equal to columns count in `cols`.